packages/expo-apple-authentication/ios/EXAppleAuthentication/EXAppleAuthentication.m

```diff
@@ -0,0 +1,200 @@
+                        rejecter:(UMPromiseRejectBlock)reject)
+{
+  if (@available(iOS 13.0, *)) {
+    _promiseResolve = resolve;
+    _promiseReject = reject;
+    
+    ASAuthorizationAppleIDProvider* appleIDProvider = [[ASAuthorizationAppleIDProvider alloc] init];
+    ASAuthorizationAppleIDRequest* request = [appleIDProvider createRequest];
+
+    NSArray<NSNumber *> *requestedScopes = options[@"requestedScopes"];
+    NSNumber *requestedOperation = options[@"requestedOperation"];
+
+    request.requestedScopes = [EXAppleAuthenticationMappings importScopes:requestedScopes];
+    request.requestedOperation = [EXAppleAuthenticationMappings importOperation:requestedOperation];
+
+    if (options[@"user"]) {
+      request.user = options[@"user"];
+    }
+    if (options[@"state"]) {
+      request.state = options[@"state"];
+    }
+    
+    ASAuthorizationController* ctrl = [[ASAuthorizationController alloc] initWithAuthorizationRequests:@[request]];
+    ctrl.presentationContextProvider = self;
+    ctrl.delegate = self;
+    [ctrl performRequests];
+  } else {
+    reject(@"ERR_APPLE_AUTHENTICATION_UNAVAILABLE", @"This feature is not available on your iPhone.", nil);
+  }
+}
+
+UM_EXPORT_METHOD_AS(getCredentialStateAsync,
+                    getCredentialStateAsync:(NSString *)userID
+                                   resolver:(UMPromiseResolveBlock)resolve
+                                   rejecter:(UMPromiseRejectBlock)reject)
+{
+  if (@available(iOS 13.0, *)) {
+    ASAuthorizationAppleIDProvider* appleIDProvider = [[ASAuthorizationAppleIDProvider alloc] init];
+    [appleIDProvider getCredentialStateForUserID:userID
+                                      completion:^(ASAuthorizationAppleIDProviderCredentialState credentialState,
+                                                   NSError * _Nullable error) {
+      if (error) {
+        return reject(@"ERR_APPLE_AUTHENTICATION", [error localizedDescription], nil);
+      }
+      resolve([EXAppleAuthenticationMappings exportCredentialState:credentialState]);
+    }];
+  } else {
+    reject(@"ERR_APPLE_AUTHENTICATION_UNAVAILABLE", @"This feature is not available on your iPhone.", nil);
+  }
+}
+
+- (ASPresentationAnchor)presentationAnchorForAuthorizationController:(ASAuthorizationController *)controller API_AVAILABLE(ios(13.0))
+{
+  return [[[UIApplication sharedApplication] delegate] window];
+}
+
+- (void)authorizationController:(ASAuthorizationController *)controller
+   didCompleteWithAuthorization:(ASAuthorization *)authorization API_AVAILABLE(ios(13.0))
+{
+  ASAuthorizationAppleIDCredential* credential = authorization.credential;
+  NSDictionary* user = @{
+                         @"fullName": [self _serializeFullName:credential.fullName],
+                         @"email": UMNullIfNil(credential.email),
+                         @"user": credential.user,
+                         @"authorizedScopes": credential.authorizedScopes,
```

Comment:
let's just leave this out -- seems like no data is ever passed back in here, and it's not of much use anyway (the application should know which scopes it requested). That avoids needing to "export" the values.